libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -326,3 +326,58 @@ AP_AHRS_View *AP_AHRS::create_view(enum Rotation rotation)
     _view = new AP_AHRS_View(*this, rotation);
     return _view;
 }
+
+/*
+ * Update AOA and SSA estimation based on airspeed, velocity vector and wind vector
+ *
+ * Based on:
+ * "On estimation of wind velocity, angle-of-attack and sideslip angle of small UAVs using standard sensors" by
+ * Tor A. Johansen, Andrea Cristofaro, Kim Sorensen, Jakob M. Hansen, Thor I. Fossen
+ *
+ * "Multi-Stage Fusion Algorithm for Estimation of Aerodynamic Angles in Mini Aerial Vehicle" by
+ * C.Ramprasadh and Hemendra Arya
+ *
+ * "ANGLE OF ATTACK AND SIDESLIP ESTIMATION USING AN INERTIAL REFERENCE PLATFORM" by
+ * JOSEPH E. ZEIS, JR., CAPTAIN, USAF
+ */
+void AP_AHRS::update_AOA_SSA(void)
+{
+    Vector3f aoa_velocity, aoa_wind;
+
+    // get velocity and wind
+    if (get_velocity_NED(aoa_velocity) == false) {
+        return;
+    }
+
+    float vel_len = aoa_velocity.length();
+
+    // do not calculate if speed is too low
+    if (vel_len < 2.0) {
+        _AOA = 0;
+        _SSA = 0;
+        return;
+    }
+
+    aoa_wind = wind_estimate();
+
+    // Rotate vectors to the body frame and calculate velocity and wind
+    Matrix3f rot = get_rotation_body_to_ned();
+    aoa_velocity = rot.mul_transpose(aoa_velocity);
+    aoa_wind = rot.mul_transpose(aoa_wind);
+
+    // calculate relative velocity in body coordinates
+    aoa_velocity = aoa_velocity - aoa_wind;
+
+    // Calculate AOA and SSA
+    if (aoa_velocity.x > 0) {
+        _AOA = degrees(atanf(aoa_velocity.z / aoa_velocity.x));
+    } else {
+        _AOA = 0;
+    }
+
+    if (!is_zero(vel_len)) {
```

Comment:
better to use (vel_len > 0) like the (aoa_vel.x > 0) above it? I know you check for (vel_len > 2) above on line 355 but you might as well stay consistent.